### PR TITLE
addressing github issue:65

### PR DESCRIPTION
**Pull request by ggfan**
Fix issue:
     https://github.com/googlesamples/android-audio-high-performance/issues/65
  Test: pass on Pixel XL device
 @philburk @dturner  PTAL.

### CODE REVIEW

File: aaudio/echo/src/main/cpp/echo_audio_engine.cc [hunk 1]
```diff
@@ -123,11 +123,13 @@ void EchoAudioEngine::stopStreams() {
   if (recordingStream_ != nullptr) {
```
Comment (philburk) [inline aaudio/echo/src/main/cpp/echo_audio_engine.cc]: The destructor also needs to either check if the pointers are null before closing,
or just call stopStreams(),
or assert that the pointers are null.

Comment (ggfan) [inline aaudio/echo/src/main/cpp/echo_audio_engine.cc]: I see:  you mean AAudioStream_close() could not take nullptr, correct?  that makes sense. let me add that one in.  If AAudioStream_close() could take a nullptr and do nothing, that would be good. I thought it as the memory function like free(), delete, I think they could take a nullptr.  let me add the patch first, thanks

File: aaudio/echo/src/main/cpp/echo_audio_engine.cc [hunk 2]
```diff
@@ -123,11 +123,13 @@ void EchoAudioEngine::stopStreams() {
...
     stopStream(recordingStream_);
     closeStream(recordingStream_);
+    recordingStream_ = nullptr;
   }
 
   if (playStream_ != nullptr) {
     stopStream(playStream_);
     closeStream(playStream_);
+    playStream_ = nullptr;
```
Comment (philburk) [inline aaudio/echo/src/main/cpp/echo_audio_engine.cc]: I see now that closeStream() also checks for null. So the destructor seems safe.

Comment (ggfan) [inline aaudio/echo/src/main/cpp/echo_audio_engine.cc]: added more comment to closeStream() function: pointing out the fact that AAudioStream_close() already takes good care of the synchronization between datacallback and close() function.  PTAL, thanks

**Pull request merged**